File: Sheet 6 (Complexity Analysis)/I.py
t = int(input())
l=[]
for _ in range(t):
    n, d = map(int, input().split())
    a = list(map(int, input().split()))
    a.sort()
    
    if a[-1] <= d:
        l.append("YES")
    elif a[0] + a[1] <= d:
        l.append("YES")
    else:
        l.append("NO")
for i in l:
    print(i)

File: Sheet 6 (Complexity Analysis)/I.cpp
#include <iostream>
#include <algorithm>
using namespace std;
int main(){
    int t;
    cin >> t;
    while (t--){
        int n,d;
        cin >>  n >> d;
        int arr[n];
        for (int i=0;i<n;i++) cin >> arr[i];
        if (*max_element(arr,arr+n)>d){
            sort(arr,arr+n);
            if (arr[0]+arr[1]>d) cout << "NO\n";
            else cout << "YES\n";
        } 
        else cout << "YES\n";
    }
}
